news/5239.bugfix.rst

```diff
@@ -0,0 +1 @@
+Sorting ``constraints`` before resolving, which fixes ``pipenv lock`` generates nondeterminism environment markers.
```

Comment:
Please rename this file to 5299.bugfix.rst to avoid the merge conflict.

Reply:
Sorry, I didn't check that. Fixed.